prov/tcp/src/tcpx_cq.c

```diff
@@ -94,6 +94,9 @@ void tcpx_pe_entry_release(struct tcpx_pe_entry *pe_entry)
 
 	tcpx_cq = container_of(cq, struct tcpx_cq, util_cq);
 
+	if (pe_entry->ep->cur_rx_entry == pe_entry)
+		pe_entry->ep->cur_rx_entry = NULL;
```

Comment:
This sort of change should be protected by some sort of lock.

Reply:
Yes it is already protected by queue_lock from this calling function tcpx_progress()